### DISCUSSION TITLE
how to deal with CORS-RFC1918

### DISCUSSION THREAD

**Discussion by zinginging** [keycloak/keycloak#12983] [Q&A]
keycloak-js got cors error as Access to XMLHttpRequest at 'http://xxxx:8080/auth/realms/tg_test/protocol/openid-connect/token' from origin 'http://0.0.0.0:6600' has been blocked by CORS policy: The request client is not a secure context and the resource is in more-private address space `private`.

**Comment by GretUp** [comment]
I am getting a similar issue:
Access to XMLHttpRequest at '[https://keycloakx.example.com/realms/xxxx/protocol/openid-connect/token'](https://keycloakx.example.com/realms/xxxx/protocol/openid-connect/token%27) from origin 'http://localhost:4200' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Related to this I found this thread: https://github.com/mauriciovigolo/keycloak-angular/issues/260

It basically says that the new javascript adapter does not support "confidential" client type. What if we need the authorization part of the client? Apparently there is no solution on this.
